From 578771947bcb024e3e1e499bf4a0e89b821869d5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Revol?= <revol@free.fr>
Date: Tue, 10 Aug 2021 16:20:25 +0200
Subject: [PATCH 1/2] Fallback to uname -o for systems without /etc/issue

This includes Haiku.

Not necessary to build but it removes an error message.
---
 src/global.pri | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/global.pri b/src/global.pri
index 4deccb0..bebbbb4 100644
--- a/src/global.pri
+++ b/src/global.pri
@@ -9,7 +9,7 @@ DEFINES += VERSION=\\\"$$VERSION\\\"
 
 # Define 'MAEMO' specific CONFIG/DEFINE
 # To handle 'MAEMO' specific soruces/code
-DISTRO = $$system(cat /etc/issue)
+DISTRO = $$system(cat /etc/issue 2>/dev/null || uname -o)
 contains( DISTRO, [Mm]aemo ) {
     # for 'MAEMO' specific source code parts
     DEFINES += MAEMO
-- 
2.30.2


From daef302a835b76fba77d3381fe0d6e3234a6dcd2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Revol?= <revol@free.fr>
Date: Tue, 10 Aug 2021 16:54:35 +0200
Subject: [PATCH 2/2] Allow overriding PREFIX

Haiku requires a different install path.
---
 src/app/app.pro | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/app/app.pro b/src/app/app.pro
index 37f4f48..b268734 100644
--- a/src/app/app.pro
+++ b/src/app/app.pro
@@ -32,7 +32,9 @@ RESOURCES += ../icons.qrc \
 			 ../../data/data.qrc
 
 # instalation related
-PREFIX = /usr/bin
+!defined(PREFIX, var) {
+    PREFIX = /usr/bin
+}
 INSTALLS = target
 target.path = $$PREFIX
 
-- 
2.30.2

